python/mxnet/runtime.py

```diff
@@ -28,21 +29,48 @@ class LibFeature(ctypes.Structure):
     Compile time feature description
     """
     _fields_ = [
-        ("name", ctypes.c_char_p),
+        ("_name", ctypes.c_char_p),
         ("index", ctypes.c_uint32),
         ("enabled", ctypes.c_bool)
     ]
 
+    @property
+    def name(self):
+        return self._name.decode()
+
+    def __repr__(self):
+        if self.enabled:
+            return "✔ {}".format(self.name)
+        else:
+            return "✖ {}".format(self.name)
+
 def libinfo_features():
     """
     Check the library for compile-time features. The list of features are maintained in libinfo.h and libinfo.cc
 
     Returns
     -------
-    A list of class LibFeature indicating which features are available and enabled
+    :return: list of class LibFeature indicating which features are available and enabled
     """
     lib_features = ctypes.POINTER(LibFeature)()
     lib_features_size = ctypes.c_size_t()
     check_call(_LIB.MXLibInfoFeatures(ctypes.byref(lib_features), ctypes.byref(lib_features_size)))
     feature_list = [lib_features[i] for i in range(lib_features_size.value)]
     return feature_list
+
+def is_enabled(tocheck):
+    """
+    Check for a particular feature by name
+
+    Parameters
+    ----------
+    :param x: str The name of a valid feature as string for example 'CUDA'
+
+    Returns
+    -------
+    :return: bool True if it's enabled, False if it's disabled, RuntimeError if the feature is not known
+    """
+    feature_dict = {f.name: f.enabled for f in libinfo_features()}
+    if tocheck not in feature_dict:
+        raise RuntimeError("Feature '{}' is unknown, known features are: {}".format(tocheck, list(feature_dict.keys())))
+    return feature_dict[tocheck]
```

Comment:
if a dictionary is easier to operate, should the libinfo_features return a dictionary?

Reply:
I prefer to keep ordering as in a list. Going from list to dict is oneliner.

Reply:
is the order information useful? I always thought of it as a set. also, by defining your own data structure class, you can define more pythonic behaviors such as `__contains__` and `__getitem__` to make it easier for feature check.

Reply:
They are grouped if you see the enum definition as some features are related, semantically is a set as you describe. For using __contains__ as you propose we would have to wrap the list in a containing class. I don't see a lot of value on complicating this further TBH. Could you provide an example of what you have in mind if you feel strongly about it?

Reply:
I don't feel stongly about the change. What I had in mind was something like:
```python
>>> features = mx.runtime.libinfo_features()
>>> 'NCCL' in features
True
>>> 'nccl' in features
True
>>> nccl_idx = 2;
>>> nccl_idx in features
True
>>> features['NCCL']
✔ NCCL
>>> features['NCCL'].index
2
>>> features['NCCL'].enabled
True
```

Reply:
I see, that looks nice. On the upside repeated calls are more efficient than my proposal. On the downside I think the primary case which is to check if a feature is enabled is less ergonomic due to the syntactic sugar. I think a simple function with a single arg is always clearer.  A good point is that it should be case insensitive. I'm a bit torn.

Reply:
Just curious is the index information useful to user? Why do we even need to expose index?

Reply:
@apeforest I just removed it.